misc/deleteFilesByQuery.py

```diff
@@ -0,0 +1,171 @@
+#
+#  IBM_PROLOG_END_TAG
+import argparse
+import json
+import sys
+import vapi
+import logging as logger
+from concurrent.futures import ThreadPoolExecutor
+from concurrent.futures import as_completed
+
+args = None
+server = None
+
+
+def main(params):
+    """ Example script to delete multiple files based upon query criteria.
+The number of files matching the criteria will be  presented and the
+user will be  prompted to perform the delete.
+A "force" flag is provided to avoid the prompt.
+    """
+    global args
+    args = getValidInputs()
+    if args is None:
+        exit(1)
+
+    setupLogging(args.log)
+    setupServer()
+    listOfFiles = getFileList()
+    deleteFilesInList(listOfFiles)
+
+
+def setupLogging(log):
+    """ Sets up the logging service."""
+
+    log_level = logger.WARNING
+
+    if log is not None:
+        # Translate log level string to logger values
+        if log.lower() == "error":
+            log_level = logger.ERROR
+        elif log.lower().startswith("warn"):
+            log_level = logger.WARNING
+        elif log.lower() == "info":
+            log_level = logger.INFO
+        elif log.lower() == "debug":
+            log_level = logger.DEBUG
+
+        logger.basicConfig(format='%(asctime)s.%(msecs)d %(levelname)s: %(message)s',
+                            datefmt='%H:%M:%S', level=log_level)
+
+
+def setupServer():
+    """ Setup the connection to the MVI Server."""
+    global server
+    try:
+        server = vapi.connect_to_server(args.baseurl, args.token)
+    except Exception as e:
+        print("Error: Failed to setup server.", file=sys.stderr)
+        logger.debug(dict(e))
+        exit(2)
+
+
+def getFileList():
+    """ Query MVI server for the list of files matching the given query."""
+    files = server.files.report(args.dsid, query=args.queryString)
+    if not server.rsp_ok():
+        print(f"Error: Got status code {server.status_code()} from file lookup. Reason = {json.dumps(server.json(), indent=2)}")
+        exit(2)
+    return files
+
+
+def deleteFilesInList(fileList):
+    """ Delete all files from the MVI server that are in the provided list of files."""
+
+    numberOfFiles = len(fileList)
+    logger.debug(f"deleting {numberOfFiles} files from dataset id {args.dsid}.")
+    if numberOfFiles < 0:
+        logger.info("Query found no files to delete.")
+        return
+    if numberOfFiles > 1000:
+        numberOfThreads = 16
+    else:
+        numberOfThreads = 8
+
+    if not okToDeleteFiles(fileList):
+        return
+    logger.info(f"Deleting {numberOfFiles} files from dataset id {args.dsid}.")
+    with ThreadPoolExecutor(max_workers=numberOfThreads) as executor:
+        results = executor.map(deleteFile, fileList)
+
+
+def deleteFile(fileObj):
+    pserver = vapi.connect_to_server(args.baseurl, args.token)
+    fileid = fileObj.get("_id", "missingId")
+    logger.debug(f"Deleting file id {fileid}")
+
+    pserver.files.delete(args.dsid, fileid)
```

Comment:
Why not use `/files/action` to do a batch delete? Running ~12k deletes takes a few minutes, even when doing parallel deletes. This is functional but slow at scale. :-)